packages/run-protocol/src/vaultFactory/vault.js

```diff
@@ -147,50 +161,44 @@ export const makeInnerVault = (
   const assertVaultIsOpen = () => {
     assertPhase(VaultPhase.ACTIVE);
   };
+
   // #endregion
 
-  let outerUpdater;
+  // CONSTANTS
 
-  // vaultSeat will hold the collateral until the loan is retired. The
-  // payout from it will be handed to the user: if the vault dies early
-  // (because the vaultFactory vat died), they'll get all their
-  // collateral back. If that happens, the issuer for the RUN will be dead,
-  // so their loan will be worthless.
-  const { zcfSeat: vaultSeat } = zcf.makeEmptySeatKit();
+  const collateralBrand = state.manager.getCollateralBrand();
+  const { brand: runBrand } = state.runMint.getIssuerRecord();
 
-  /**
-   * Snapshot of the debt and compounded interest when the principal was last changed
-   *
-   * @type {{run: Amount, interest: Ratio}}
-   */
-  let debtSnapshot = {
-    run: AmountMath.makeEmpty(runBrand, 'nat'),
-    interest: manager.getCompoundedInterest(),
-  };
+  let outerUpdater;
 
   /**
    * Called whenever the debt is paid or created through a transaction,
    * but not for interest accrual.
    *
-   * @param {Amount} newDebt - principal and all accrued interest
+   * @param {Amount<NatValue>} newDebt - principal and all accrued interest
```

Comment:
<NatValue> is going to get tiresome real quick, given that it's pervasive for fungible tokens.